core/src/smartcontracts/isi/query.rs

```diff
@@ -153,19 +153,22 @@ impl Error {
     pub const fn status_code(&self) -> StatusCode {
         use Error::*;
         match *self {
-            Conversion(_) | Decode(_) | Version(_) => StatusCode::BAD_REQUEST,
+            Decode(_) | Version(_) | Evaluate(_) | Conversion(_) => StatusCode::BAD_REQUEST,
             Signature(_) => StatusCode::UNAUTHORIZED,
-            Evaluate(_) | Permission(_) | Find(_) => StatusCode::NOT_FOUND,
+            Permission(_) => StatusCode::FORBIDDEN,
```

Comment:
Yeah, that's exactly why I previously gave `NOT_FOUND` to `Permission` error, following the practice in Web servers.
I think `FORBIDDEN` is ok because it never informs the existence of the resource to attackers
since the permission check precedes in our API, contrary to the case of ordinary Web pages

Reply:
Oh, I see, thanks